core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/kerberos/KeytabProvider.java

```diff
@@ -26,7 +26,12 @@ public ServiceKeytabResponse getServiceKeytabResponse(Stack stack, GatewayConfig
             throws Exception {
         ServiceKeytabRequest request = new ServiceKeytabRequest();
         request.setEnvironmentCrn(stack.getEnvironmentCrn());
-        request.setServerHostName(primaryGatewayConfig.getHostname());
+        String hostname = primaryGatewayConfig.getHostname();
+        request.setServerHostName(hostname);
+        int dotIndex = hostname.indexOf('.');
+        if (dotIndex != -1) {
+            request.setServiceAlias(hostname.substring(0, dotIndex));
+        }
```

Comment:
StringUtils#substringBefore would be simpler: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringBefore-java.lang.String-java.lang.String-

Reply:
Makes sense, and I will add some unit test